File: 2024/tinhoclk/tower.py
import math
test = int(input())
for i in range(test) : 
    x = int(input()) 
    delta =  1+8*x  
    b = int(math.sqrt(delta)) 
    if b * b == delta and (-1+b)%2==0:
        print("YES") 
    else : 
        print("NO")  

File: 2024/tinhoclk/tower.cpp
#include<bits/stdc++.h>
using namespace std ;

signed main()
{
    int test ;cin>>test ; 
    while(test--)
    {
        long long val; cin>>val;  
        long long delta = 1+8*val;
        long long b = sqrt(delta)  ; 
        if(b*b==delta && (-1+b)%2==0)
        {
            cout<<"YES"<<'\n';
        }
        else 
        {
            cout<<"NO"<<'\n' ;
        }
    }
}